Updated to zig 0.14.1

Yinameah opened the pull request:
Hello. 

I was toying around with the wasm example, trying to solve a weird bug I have in my project,  and since I did change the stuff to build for latest zig, I figured I would share.

Took the opportunity to copy the assets to a single `web` folder, this makes a nice live workflow with `zig build --watch`

Feel free to merge if you see fit or to close the PR otherwise.

Great work on zf btw, love the simplicity.

Yinameah commented:
Oulà ! wait. The weird ranking bug I was chasing is actually due to the 0.14 change.

The left/white one is working on this commit on 0.14 and latest zf release, while the right/black one is the online version. Something is clearly broken. Not sure where and why. 
![image](https://github.com/user-attachments/assets/82622695-577f-4f53-9c98-aa9a46ca77ac)

I did not noticed before because with the built-in examples I could not spot something weird.

natecraddock commented:
Thanks! I'll take a closer look later

> weird ranking bug I was chasing is actually due to the 0.14 change

Does this bug show in zf? Or only in the wasm playground?

Yinameah commented:
Writing a follow up in the main repo as we speak ;-)

Yinameah commented:
Oookay. So this is what I was seeing :

<image src="https://github.com/user-attachments/assets/faae7859-1dc7-4a45-a89d-21b418021e83" width=450>

Should I have added the rank in the browser earlier, I would have spotted earlier than the issue was with java script sorting. (and that **zf** means the lower the better).

And the root cause is, I guess, javascript being inconsistently typed ? Because this issue was only apparent in chromium, but when I switched to firefox it was fine.

And it seems like the proper way to sort stuff is with `sort((a,b) => a - b)` and **not** with `sort((a,b) => a > b)`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

And so `false -> 0 => no sort swap` hence the result is initial order dependent, hence I was very confused. Don't ask me why firefox don't comply to their own spec and chromium does.

natecraddock commented:
Thanks for digging into this and fixing the bug! So weird that this would be a chromium / firefox difference. I'll test it later to confirm that everything is working then I'll merge

Yinameah commented:
Yeah, please double check. This is quite unexpected .

natecraddock commented:
Tested and things are working well! Deployed and the updates are live on https://nathancraddock.com/zf-playground/

Thank you so much again for catching this. It's a bit embarrassing to know that people testing the playground on Chromium based browsers didn't get a great experience!
